Initiate charm modernisation

jat-canonical opened the pull request:
This commit adds basic configuration to linters and formatters. This commit also satisfies the linters and formatters. This commit also creates jobs for github actions.

jat-canonical committed:
feat(charms): add basic configurations for linter & formatters

This commit adds basic configuration to linters and formatters. This
commit also satisfies the linters and formatters.

jat-canonical committed:
feat(charms): drop bionic from charmcraft

As we have stopped supporting bionic we should stop building a charm
for it.

jat-canonical committed:
feat(tests): add unit and integration tests

This commit adds some basic unit and integration tests using Harness and
the pytest-operator plugin respectively. This also adds support for
testing against both juju 3.2 and juju 2.9.

morphis commented on .github/renovate-config.js:
```diff
@@ -0,0 +1,10 @@
+  onboarding: true,
+  platform: "github",
+  includeForks: true,
+  repositories: ["jat-canonical/nats-operator"],
```
This is wrong and needs to become "canonical/charm-lxd-integrator"

morphis commented on .github/workflows/build-and-test.yaml:
```diff
@@ -0,0 +1,58 @@
+
+  integration-test:
+    name: Integration tests
+    runs-on: ubuntu-20.04
```
Why 20.04? Lets use 22.04 if there is nothing speaking against it.

jat-canonical replied:
the charm supports building on 20.04 to support running on 20.04 as well. Hence had to run the integration tests on 20.04.

jat-canonical commented:
@morphis Have a look at this one. Not sure why the GHA aren't running.

morphis reviewed (changes requested):
Cam we update README and CONTRIBUTING as well as we did for the nats-operator charm?

jat-canonical committed:
feat(docs): Add CONTRIBUTING.md and update README.md with the new format

This commit updates the readme to updated format and add the
contribution guidelines to the repository.

morphis commented on README.md:
```diff
@@ -64,4 +71,13 @@ protocol, host and port.
+Security issues in the operator can be reported through [LaunchPad](https://wiki.ubuntu.com/DebuggingSecurity#How%20to%20File) on the [Anbox Cloud](https://bugs.launchpad.net/anbox-cloud) project. Please do not file GitHub issues about security issues.
+
+## Contributing
+Please see the [Juju SDK docs](https://juju.is/docs/sdk) for guidelines on enhancements to this charm following best practice guidelines, and [CONTRIBUTING.md](https://github.com/canonical/nats-operator/blob/main/CONTRIBUTING.md) for developer guidance.
```
This is linking to the CONTRIBUTING.md of the nats-operator. We need to have our own here and link it.

morphis commented on README.md:
```diff
@@ -64,4 +71,13 @@ protocol, host and port.
+Security issues in the operator can be reported through [LaunchPad](https://wiki.ubuntu.com/DebuggingSecurity#How%20to%20File) on the [Anbox Cloud](https://bugs.launchpad.net/anbox-cloud) project. Please do not file GitHub issues about security issues.
+
+## Contributing
+Please see the [Juju SDK docs](https://juju.is/docs/sdk) for guidelines on enhancements to this charm following best practice guidelines, and [CONTRIBUTING.md](https://github.com/canonical/charm-lxd-integrator/blob/main/CONTRIBUTING.md) for developer guidance.
```
We still miss the actual CONTRIBUTING.md file in the PR

morphis replied:
Let me also rename the branch to "main" so the link actually works.

morphis reviewed (approved):
Going to merge this as-is and add the `CONTRIBUTING.md` myself to see why we don't get the actions to work.
